test/e2e/multiple_namespaces_test.go

```diff
@@ -0,0 +1,129 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/vmware-tanzu/velero/pkg/util/kube"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var _ = Describe("[Basic] Backup/restore of 2 namespaces", func() {
+	var (
+		client *kubernetes.Clientset
+	)
+
+	BeforeEach(func() {
+		var err error
+		client, _, err = kube.GetClusterClient()
+		Expect(err).To(Succeed(), "Failed to instantiate cluster client")
+	})
+
+	Context("When I create 2 namespaces", func() {
+		It("should be successfully backed up and restored", func() {
+			backupName := "backup-" + uuidgen.String()
+			restoreName := "restore-" + uuidgen.String()
+			fiveMinTimeout, _ := context.WithTimeout(context.Background(), 5*time.Minute)
+			RunMultipleNamespaceTest(fiveMinTimeout, client, "nstest-"+uuidgen.String(), 2,
+				backupName, restoreName)
+		})
+	})
+})
+
+var _ = Describe("[Scale] Backup/restore of 2500 namespaces", func() {
+	var (
+		client *kubernetes.Clientset
+	)
+
+	BeforeEach(func() {
+		var err error
+		client, _, err = kube.GetClusterClient()
+		Expect(err).To(Succeed(), "Failed to instantiate cluster client")
+	})
+
+	Context("When I create 2500 namespaces", func() {
+		It("should be successfully backed up and restored", func() {
+			backupName := "backup-" + uuidgen.String()
+			restoreName := "restore-" + uuidgen.String()
+			oneHourTimeout, _ := context.WithTimeout(context.Background(), 1*time.Hour)
+			RunMultipleNamespaceTest(oneHourTimeout, client, "nstest-"+uuidgen.String(), 2500,
+				backupName, restoreName)
+		})
+	})
+})
+
+func RunMultipleNamespaceTest(ctx context.Context, client *kubernetes.Clientset, nsBaseName string, numberOfNamespaces int, backupName string, restoreName string) error {
+	shortTimeout, _ := context.WithTimeout(ctx, 5*time.Minute)
+	defer cleanupNamespaces(ctx, client, nsBaseName) // Run at exit for final cleanup
+	var excludeNamespaces []string
+
+	// Currently it's hard to build a large list of namespaces to include and wildcards do not work so instead
+	// we will exclude all of the namespaces that existed prior to the test from the backup
+	namespaces, err := client.CoreV1().Namespaces().List(shortTimeout, v1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "Could not retrieve namespaces")
+	}
+
+	for _, excludeNamespace := range namespaces.Items {
+		excludeNamespaces = append(excludeNamespaces, excludeNamespace.Name)
+	}
+	for nsNum := 0; nsNum < numberOfNamespaces; nsNum++ {
+		createNSName := fmt.Sprintf("%s-%00000d", nsBaseName, nsNum)
+		if err := createNamespace(ctx, client, createNSName); err != nil {
+			return errors.Wrapf(err, "Failed to create namespace %s to install Kibishii workload", kibishiiNamespace)
```

Comment:
I'd remove the reference to Kibishii here as we aren't installing any workloads in these namespaces.

Reply:
Mmmm...you're right, it's actually using the wrong namespace of what failed here